feat: Integrate Ktlint for Code Style Enforcement

Alirezanr opened the pull request:
This pull request integrates Ktlint into the project to enforce consistent code style across the codebase.

    Key changes:
    - Added the Ktlint Gradle plugin.
    - Configured Ktlint rules in `build.gradle.kts`.
    - Applied initial Ktlint formatting to existing code.
    - Added `ktlintFormat` and `ktlintCheck` tasks.

    This will help maintain code quality and readability.

Alirezanr committed:
#8 Added ktlint plugin with configuration.

Alirezanr committed:
#8 Fixed project structure and code style using the ktlint checks.

Alirezanr committed:
Merge branch 'develop' into alirezanr/add-ktlint

# Conflicts:
#	gradle/libs.versions.toml

viamgr commented on .editorconfig:
```diff
@@ -0,0 +1,24 @@
+end_of_line = lf
+insert_final_newline = false
+trim_trailing_whitespace = true
+max_line_length = 160
```
I recommend to use lower numbers for max line length

Alirezanr replied:
Is 120 chars ok?

viamgr replied:
120 is ok.

viamgr commented on app/build.gradle.kts:
```diff
@@ -1,8 +1,11 @@
     id(libs.plugins.kotlin.android.get().pluginId) 
     alias(libs.plugins.kotlin.compose)
     alias(libs.plugins.kotlin.ksp) apply false
+    alias(libs.plugins.ktlint)
```
Doesn't it needs to apply this plugin and it's configuration to the android module plugins that we have already defined?

mmbehroozfar replied:
Don't we need to apply it in project level Gradle file?

Alirezanr replied:
No, I tested and the ktlint rules apply to all the project files.

Alirezanr committed:
#8 !24 refactor: Move Ktlint configuration and preBuild task dependency to the `AndroidModulePlugin`.

Alirezanr committed:
#8 !24 refactor: Format project files based on the ktlint rules.

Alirezanr committed:
#8 !24 refactor: Update max_line_length from 160 to 120 in .editorconfig.

MohammadBagherNoorian commented on .editorconfig:
```diff
@@ -0,0 +1,24 @@
+[*.{kt,kts}]
+ij_kotlin_imports_layout = *,java.**,javax.**,kotlin.**,^
+ktlint_standard_no-wildcard-imports = enabled
+ktlint_standard_final-newline = disabled
```
Please consider changing `insert_final_newline = false` to `true` as:
   - It follows POSIX standards
   - Prevents issues with git diffs and file concatenation
   - Most tools expect files to end with a newline
   https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreeol
   https://pinterest.github.io/ktlint/1.5.0/rules/standard/

Alirezanr replied:
Good point!

viamgr committed:
Merge branch 'develop' of https://github.com/epic-labs-org/ThinkTokAndroid into alirezanr/add-ktlint

# Conflicts:
#	app/build.gradle.kts
#	gradle/libs.versions.toml

viamgr committed:
Merge branch 'alirezanr/add-ktlint' of https://github.com/epic-labs-org/ThinkTokAndroid into alirezanr/add-ktlint-fix

# Conflicts:
#	app/build.gradle.kts

viamgr committed:
Refactor: Consolidated common build configurations and enhanced code quality

- Renamed `common.kt` to `Common.kt` for better naming conventions.
- Added support for conditional ProGuard rules based on file existence.
- Configured debug builds to disable minification.
- Applied `ktlint` plugin to all subprojects, enabling stricter code style enforcement.
- Configured `ktlint` to check and format code, set to fail builds on errors.
- Updated test tasks to use JUnit for consistency.
- Enhanced `AppModulePlugin` and `AndroidModulePlugin` to integrate new build configurations.
- Enabled experimental rules and verbose output for `ktlint` in subprojects.
- Configured pre-build tasks to depend on `ktlintFormat`.
- Integrated the `ktlint` plugin in `buildSrc` and configured it.
- Removed the unnecessary ktlint configuration from the app module build file.

viamgr committed:
Added `.gitignore` files to `data`, `feature`, and `feature/main` modules. Each `.gitignore` file ignores the `/build` directory.

viamgr committed:
Added Git Hooks and Installation Script

- Added `install-git-hooks.sh` script to install Git hooks.
- Added `pre-push` hook in `scripts/git-hooks/` to run `ktlintFormat` before each push.
- Updated `README.md` to include instructions for setting up Git hooks.

viamgr committed:
Added TestClass with a test property to the main UI module.

viamgr committed:
Refactored TestClass and updated git hooks

- Removed unnecessary spaces in `TestClass` property declaration.
- Replaced the `pre-push` git hook with a `pre-commit` hook.

viamgr committed:
Refactored `TestClass` formatting in `main-ui` module.

viamgr committed:
Refactored `TestClass` formatting in `main-ui` module.

viamgr committed:
Refactored `TestClass` formatting in `main-ui` module.

viamgr committed:
Refactored Git hooks and Kotlin code formatting

- Renamed `pre-commit` Git hook to `pre-receive`.
- Updated `install-git-hooks.sh` to copy and set executable permissions for `pre-receive` hook.
- Removed unnecessary spaces in `TestClass.kt` property declaration.

viamgr committed:
Refactored `TestClass.kt` for code style consistency.

- Reformatted the property declaration within `TestClass` to improve code readability.

viamgr committed:
Revert "Refactored Git hooks and Kotlin code formatting"

This reverts commit 278a46a9

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Refactored `TestClass` formatting

- Removed unnecessary spaces in `TestClass`'s `testProperty` declaration.

viamgr committed:
Removed `TestClass` and `TestClass2` from the main UI module.

viamgr committed:
Refactored ktlint configuration and added it to buildSrc module

- Added ktlint format for buildSrc to the pre-commit hook.
- Removed redundant kotlin options configuration in buildSrc module.

Alirezanr committed:
Merge pull request #28 from epic-labs-org/alirezanr/add-ktlint-fix

alirezanr/add-ktlint-fix

mmbehroozfar commented on buildSrc/build.gradle.kts:
```diff
@@ -11,4 +14,13 @@ dependencies {
+    implementation("org.jlleitschuh.gradle:ktlint-gradle:12.1.2")
+}
+ktlint {
+    android = false // If you're not using Android, set to false
```
Don't we use Android?
